beacon-chain/core/electra/registry_updates.go

```diff
@@ -32,34 +33,37 @@ import (
 //	         for validator in state.validators:
 //	             if is_eligible_for_activation(state, validator):
 //	                 validator.activation_epoch = activation_epoch
-func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) (state.BeaconState, error) {
+func ProcessRegistryUpdates(ctx context.Context, state state.BeaconState) error {
 	currentEpoch := time.CurrentEpoch(state)
 	ejectionBal := params.BeaconConfig().EjectionBalance
 	activationEpoch := helpers.ActivationExitEpoch(currentEpoch)
 	vals := state.Validators()
 	for idx, val := range vals {
+		// Handle validators eligible to join the activation queue.
 		if helpers.IsEligibleForActivationQueue(val, currentEpoch) {
 			val.ActivationEligibilityEpoch = currentEpoch + 1
 			if err := state.UpdateValidatorAtIndex(primitives.ValidatorIndex(idx), val); err != nil {
-				return nil, err
+				return err
 			}
 		}
-		if helpers.IsActiveValidator(val, currentEpoch) && val.EffectiveBalance <= ejectionBal {
+		// Handle validator ejections.
+		if val.EffectiveBalance <= ejectionBal && helpers.IsActiveValidator(val, currentEpoch) {
 			var err error
-			maxExitEpoch, churn := validators.MaxExitEpochAndChurn(state) 
-			state, _, err = validators.InitiateValidatorExit(ctx, state, primitives.ValidatorIndex(idx), maxExitEpoch, churn)
+			// exitQueueEpoch and churn arguments are not used in electra.
+			state, _, err = validators.InitiateValidatorExit(ctx, state, primitives.ValidatorIndex(idx), 0 /*exitQueueEpoch*/, 0 /*churn*/)
 			if err != nil {
-				return nil, err
+				return err
```

Comment:
maybe worth adding an errors.Wrap here to know it's bubbling up for initiate validator exit